### PR TITLE
[dune] [opam] Install `revision` file when building with Dune.

### LINKED ISSUE #8621: non-release coqtop stopped reporting its current commit
#### Version

$ coqtop --version
The Coq Proof Assistant, version 8.10+alpha (October 2018)
compiled on Oct 1 2018 23:35:50 with OCaml 4.07.0

It comes from a fresh OPAM `dev` installation. Basing on the time I started installation of the switch, it is either a7ba3df749df51ffdddce055bb630346522d88b0 or 05786b23cf0d031c93998c59f6f2f94d6049b027.

#### Operating system

```shell
$ head -1 /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
```

#### Description of the problem

`echo | coqtop` used to show git commit `coqtop` was built from, but it stopped:

```
$ echo | coqtop
Welcome to Coq 8.10+alpha (October 2018)

Coq < Coq < 
```

### PR DESCRIPTION
**Pull request by ejgallego**
Fixes #8621

### CODE REVIEW

File: dev/tools/make_git_revision.sh
Comment (Zimmi48) [inline dev/tools/make_git_revision.sh]: Is bash absolutely required here? I think we have tried until then to keep the main Makefile shell agnostic (this doesn't apply to e.g. the CI or test-suite Makefile AFAIR).

File: dune
Comment (Zimmi48) [inline dune]: Where can I find documentation on `:rev-script`?

**Review by Zimmi48 — COMMENTED** [review]
Welcomed clean-up (in addition to the fix).

File: dev/tools/make_git_revision.sh
Comment (ejgallego) [reply dev/tools/make_git_revision.sh]: > Is bash absolutely required here? I think we have tried until then to keep the main Makefile shell agnostic (this doesn't apply to e.g. the CI or test-suite Makefile AFAIR).

The build system is far from agnostic to `bash`.

File: dune
Comment (ejgallego) [reply dune]: https://dune.readthedocs.io/en/latest/dune-files.html#dependency-specification

File: dev/tools/make_git_revision.sh
Comment (Zimmi48) [reply dev/tools/make_git_revision.sh]: As long as other build system maintainers are happy with this, I don't mind.

File: dune
Comment (Zimmi48) [reply dune]: Thanks.

File: dune
Comment (Zimmi48) [inline dune]: I didn't find where this was documented but is the script going to be run at every build regardless of whether `dev/tools/make_git_revision.sh` changed or not? Because if that's not the case, then the output of `coqtop | echo` might become outdated, mightn't it?

Comment (ejgallego) [reply dune]: This has roughly the same semantics than the current makefile, I think, which indeed are problematic as they can go out of date as you point out (see #483).

The file will be created once thus. That works for the OPAM build, not so well for the dev build, but this issue is global, I refer to the discussion on the several issues about versioning.

Comment (Zimmi48) [reply dune]: OK I was not aware that the make-based system suffered the same issue.

**Comment by ejgallego** [comment]
@Zimmi48 here you have another instance of a bugger `needs: rebase`. Leaving the PR untouched.

**Comment by Zimmi48** [comment]
Thanks. I'll look into this ASAP. Feel free to rebase (now there have really been updates to master in the meantime).

**Review by gares — APPROVED** [review]

**Comment by ejgallego** [comment]
Looks ready to go, assignee needed.

**Comment by Zimmi48** [comment]
@gares? You seem to me like the most important code owner here.

**Comment by ejgallego** [comment]
Ping for an assignee.

**Comment by Zimmi48** [comment]
I'll take it since it already has @gares's approval.

**Review by Zimmi48 — APPROVED** [review]

**Comment by ejgallego** [comment]
Merci @Zimmi48

**Pull request merged**
